components/telemetry-operator/internal/webhook/dryrun/file_writer.go

```diff
@@ -0,0 +1,164 @@
+package dryrun
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+	"github.com/kyma-project/kyma/components/telemetry-operator/internal/fluentbit"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+//go:generate mockery --name fileWriter --filename file_writer.go
+type fileWriter interface {
+	preparePipelineDryRun(ctx context.Context, workDir string, pipeline *telemetryv1alpha1.LogPipeline) (func(), error)
+	prepareParserDryRun(ctx context.Context, workDir string, pipeline *telemetryv1alpha1.LogParser) (func(), error)
+}
+
+type fileWriterImpl struct {
+	client client.Client
+	config *Config
+}
+
+func (f *fileWriterImpl) prepareParserDryRun(ctx context.Context, workDir string, parser *telemetryv1alpha1.LogParser) (func(), error) {
+	if err := makeDir(workDir); err != nil {
+		return nil, err
+	}
+	if err := f.writeConfig(ctx, workDir); err != nil {
```

Comment:
Is the config even necessary when validating the parsers files? Executing `fluent-bit --dry-run --parser=./custom_parsers.conf --parser=./dynamic-parsers/parsers.conf` shows me errors in the parsers files.

Reply:
Nope, got removed while you were reviewing 😁